addons/project/migrations/10.0.1.1/pre-migration.py

```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+
+
+# Delete XML IDs
+@openupgrade.migrate(use_env=True)
+def migrate(env, version):
+    cr = env.cr
+    env.ref('project.portal_project_rule').unlink()
+    env.ref('project.portal_task_rule').unlink()
+    env.ref('project.action_client_project_menu').unlink()
+    cr.execute("""DELETE FROM
```

Comment:
those are actually not necessary, because the unlinks above will also clean up xmlids pointing to the records in question: https://github.com/OCA/OCB/blob/10.0/odoo/models.py#L3418-L3420
And xmlids not in use any more are also cleaned up by the system